runtime/Makefile

```diff
@@ -671,6 +693,53 @@ $(foreach platform,$(DOTNET_PLATFORMS),$(foreach rid,$(DOTNET_$(platform)_RUNTIM
 $(foreach platform,$(DOTNET_PLATFORMS),$(foreach rid,$(DOTNET_$(platform)_RUNTIME_IDENTIFIERS),$(foreach arch,$(DOTNET_$(rid)_ARCHITECTURES),$(eval $(call DotNetLibXamarinTemplate,$(DOTNET_$(rid)_SDK_PLATFORM),$(rid),$(arch),_CORECLR,-dotnet-coreclr,_CORECLR)))))
 $(foreach platform,$(DOTNET_PLATFORMS),$(foreach rid,$(DOTNET_$(platform)_RUNTIME_IDENTIFIERS),$(foreach arch,$(DOTNET_$(rid)_ARCHITECTURES),$(eval $(call DotNetLibXamarinTemplate,$(DOTNET_$(rid)_SDK_PLATFORM),$(rid),$(arch),_CORECLR_DEBUG,-dotnet-coreclr-debug,_CORECLR)))))
 
+#
+# DotNetLibExtensionTemplate builds libapp.a and lib[tv]extension.a
+#
+
+define DotNetLibExtensionTemplate
+
+# TODO note that we build libextension and libtvextension
+# for all cases, even though we need at most one of them.
+# for example, the output directory for tvos contains
+# libextension-dotnet.a (which is for iOS)
+# and libtvextension-dotnet.a (which is for tvOS)
+# it seems we should only need to build one.
+
+# note also that we build libapp-dotnet, but I'm not sure
+# we need that one at all.
+
+# note also that there is a separate set of instructions to
+# build these, above, in the monotouch section of the file,
+# the instructions here build (potentially identical) files
+# except with -dotnet included in the name.
```

Comment:
Yes, the logic to build the legacy Xamarin libraries will be deleted once we drop support for legacy Xamarin. Until that happens, we're going to keep both versions around.